libcontainer/intelrdt: cleanups

kolyshkin opened the pull request:
A bunch of cleanups for the code in `libcontainer/intelrdt`, mostly to simplify or remove the code, with occasional minor improvements and one nil dereference fix. See individual commit for details.

Please review commit-by-commit.

kolyshkin commented:
@marquiz @Creatone @xiaochenshen PTAL; I'd appreciate your reviews.

xiaochenshen commented:
@kolyshkin Thank you very much for the code cleanups and bug fixing for Intel RDT.

Three tiny typos in subject of this PR, description of this PR and subject of last commit: 😃 
libcontainer/intelrtd  --> libcontainer/intelrdt

xiaochenshen commented on libcontainer/specconv/spec_linux.go:
```diff
@@ -300,22 +300,17 @@ func CreateLibcontainerConfig(opts *CreateOpts) (*configs.Config, error) {
-			}
-			if spec.Linux.IntelRdt.MemBwSchema != "" {
-				config.IntelRdt.MemBwSchema = spec.Linux.IntelRdt.MemBwSchema
+			config.IntelRdt = &configs.IntelRdt{
+				L3CacheSchema: spec.Linux.IntelRdt.L3CacheSchema,
+				MemBwSchema:   spec.Linux.IntelRdt.MemBwSchema,
```
LGTM. Thank you.

xiaochenshen commented on libcontainer/intelrdt/intelrdt.go:
```diff
@@ -547,7 +547,7 @@ func (m *intelRdtManager) Apply(pid int) (err error) {
 		return nil
 	}
 	d, err := getIntelRdtData(m.config, pid)
-	if err != nil && !IsNotFound(err) {
+	if err != nil {
```
LGTM. Thank you.

xiaochenshen reviewed (approved):
LGTM. Thank you.

xiaochenshen reviewed (dismissed):
LGTM for all 6 commits (except typos nits). Thank you.

kolyshkin commented:
> Three tiny typos in subject of this PR, description of this PR and subject of last commit: smiley
> libcontainer/intelrtd --> libcontainer/intelrdt

@xiaochenshen thanks for your review!

Fixed, and I also found a 4th one, in a commit before last, also fixed:
```diff
@@ -400,7 +400,7 @@ func writeFile(dir, file, data string) error {
                return fmt.Errorf("no such directory for %s", file)
        }
        if err := ioutil.WriteFile(filepath.Join(dir, file), []byte(data+"\n"), 0o600); err != nil {
-               return NewLastCmdError(fmt.Errorf("intelrdt: unable to write %v: %w", data, err))
+               return NewLastCmdError(fmt.Errorf("intelrtd: unable to write %v: %w", data, err))
        }
        return nil
 }
```

kolyshkin commented:
Out of curiosity, I checked the impact of this PR on runc binary size
```console
[kir@kir-rhat runc]$ size runc-v*
   text	   data	    bss	    dec	    hex	filename
7531045	3885612	 223416	11640073	 b19d09	runc-v1.0.0-38-ge5ccc4b9
7528725	3884100	 223416	11636241	 b18e11	runc-v1.0.0-44-g0229a77a
```

Almost 4K savings out of nowhere.

kolyshkin commented:
CI failure is a glitch; CI restarted

xiaochenshen reviewed (approved):
LGTM. Thank you!

Creatone commented:
LGTM. Thanks!

marquiz commented:
Thanks @kolyshkin!

LGTM

marquiz commented:
ping @crosbymichael @mrunalp @dqminh @hqhq @cyphar

I'd really love to get this in in order to be able proceed with #2920
